### PR TITLE
Feat/secondary destination

### PR DESCRIPTION
**Pull request by johpiip**
Not ready to merge yet!

This pull request will produce estimation data for modeling the choice of a secondary destination. Time-averaging of travel time and cost matrices may still change after Virpi's review but I will add commits here if needed.

Previously, I have tried to keep all datasets similar, meaning that they all go through the same scripts and the output is produced in same way. This dataset, however, is different as some of its variables, travel time and cost, are not calculated the same way as in previous datasets. Instead of calculating a travel time in tour A-B-C-A, we calculate a difference in travel times when traveling A-B-C-A instead of just A-B-A.

### CODE REVIEW

**Comment by johpiip** [comment]
This pull request is ready to merge! This PR adds the functionality to create a secondary destination estimation data, and also clears up some batch routines. I still need to update time-averaging scripts for metropolitan data but I will do it in another branch.

File: tours/tour.py [hunk 1]
```diff
@@ -192,6 +193,50 @@ def get_destination(self, origin):
...
+        locations = [location for location in self.get_locations() if
+                     location is not origin and location is not destination]
+        if not locations:
+            return empty_location
+        groups = list()
+        distances = list()
+        low_priority = max(constants.TYPE_GROUP.values()) + 100
+        for location in locations:
+            group = constants.TYPE_GROUP[location.get_type()]
+            if location is origin:
```
Comment (ktuukkan) [inline tours/tour.py]: Isn't this already guaranteed by the filtering on line 202?

File: tours/tour.py [hunk 2]
```diff
@@ -192,6 +193,50 @@ def get_destination(self, origin):
...
+                # After home, work, school, and study locations (1-4) all other
+                # location groups are of same priority.
+                groups.append(low_priority)
+                distance1 = origin.eucd(location)
+                distance2 = destination.eucd(location)
+                distances.append(constants.if_nan_then(distance1, 0.0) +
+                                 constants.if_nan_then(distance2, 0.0))
+            else:
+                groups.append(group)
+                distance1 = origin.eucd(location)
```
Comment (ktuukkan) [inline tours/tour.py]: Only `groups` seem to be affected by this if clause, so I think moving `distance1`, `distance2` and `distances` outside the clause would reduce some duplicate lines.

File: tours/tour.py [hunk 3]
```diff
@@ -192,6 +193,50 @@ def get_destination(self, origin):
         destination = locations[m]
         return destination
 
+    def get_secondary_destination(self, origin, destination, empty_location):
```
Comment (ktuukkan) [inline tours/tour.py:196]: In general, I think this function could benefit of some sort of `maxBy` function, but not sure how this works in Python.

**Review by ktuukkan — APPROVED** [review]

File: tours/tour.py (hunk 2 above)
Comment (johpiip) [reply tours/tour.py]: 3f2ac5d6a42a45fd85fbf06f69c9ccf01443f295

File: tours/tour.py (hunk 1 above)
Comment (johpiip) [reply tours/tour.py]: 3f2ac5d6a42a45fd85fbf06f69c9ccf01443f295

File: tours/tour.py (hunk 2 above)
Comment (johpiip) [reply tours/tour.py]: Thank you for your comments!

File: tours/tour.py (hunk 3 above)
Comment (johpiip) [reply tours/tour.py:196]: I will look into this later, thanks!

**Pull request merged**